### PR TITLE
[dynamo][ac] Support kwargs for activation checkpointed higher order ops

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #104577
* #104639
* #104565

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78

### CODE REVIEW

**Comment by zou3519** [comment]
I'm still reading through both PRs, but I think https://github.com/pytorch/pytorch/pull/104180 tries to do something similar? cc @kshitij12345

File: test/dynamo/test_activation_checkpointing.py [hunk 1]
```diff
@@ -281,6 +297,40 @@ def fn(x, y):
         self.assertEqual(cnt.op_count, 2)
         self.assertEqual(len(cnt.graphs), 2)
 
+    @requires_cuda()
+    def test_kwargs(self):
+        def gn(x, y, z=None):
```
Comment (zou3519) [inline test/dynamo/test_activation_checkpointing.py:302]: What prompted this PR? Did you notice a lot of modules needing to handle kwargs?

Comment (anijain2305) [reply test/dynamo/test_activation_checkpointing.py:302]: Yes, activation checkpointing for transfomers typically have forward methods with lots of kwargs.

**Comment by zou3519** [comment]
> I'm still reading through both PRs, but I think #104180 tries to do something similar? cc @kshitij12345

I read through both PRs. They're both trying to do the same thing. I think #104180 is a bit further along, so I commented there

**Pull request closed without merging**